generator/src/test/java/com/scottlogic/deg/generator/constraints/IsInSetConstraintTests.java

```diff
@@ -0,0 +1,57 @@
+package com.scottlogic.deg.generator.constraints;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.constraints.atomic.IsEqualToConstantConstraint;
+import com.scottlogic.deg.generator.constraints.atomic.IsInSetConstraint;
+import com.scottlogic.deg.generator.inputs.RuleInformation;
+import com.scottlogic.deg.generator.restrictions.NullRestrictions;
+import com.scottlogic.deg.schemas.v3.RuleDTO;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+public class IsInSetConstraintTests {
+
+    @Test
+    public void testConstraintThrowsIfGivenEmptySet(){
+        Field field1 = new Field("TestField");
+        try {
+            new IsInSetConstraint(field1, Collections.emptySet(), rules());
+        } catch (Exception e) {
+            Assert.assertThat(e, is(instanceOf(IllegalArgumentException.class)));
+            return;
```

Comment:
Could we rewrite the test so that it can be clearer what we are asserting? I think the structure is a bit hard to follow with the return in the catch and the failing of the test if we reach the end of the code block. e.g. 


    @Test
    public void testConstraintThrowsIfGivenEmptySet(){

        Field field1 = new Field("TestField");
        Assertions.assertThrows(IllegalArgumentException.class, () ->  new IsInSetConstraint(field1, Collections.emptySet(), rules()));
    }

Reply:
Changed